trucks.js

```diff
@@ -501,4 +504,68 @@ var foodTrucks = [
 		Facebook: 'https://www.facebook.com/fulltilticecreams',
 		Twitter: 'https://twitter.com/fticecream'
 	}
-];
\ No newline at end of file
+];
+
+
+// this module should support the following methods:
+// getTrucks() - return all trucks
+var getTrucks = function(){
+	return foodTrucks;
+};
+// getTruck(name) - return the truck object matching 'name'
+var getTruck = function(name){
+	return _.find(foodTrucks, function(truck){
+		return truck.name === name;
+	});
+};
+// getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+var getFoodTypes = function(){
+	var foodTypesArray = [];
+	_.each(foodTrucks, function(truck){
+		_.each(truck.type, function(truckType){
+			foodTypesArray.push(truckType);
+		});
+	});
+	return _.uniq(foodTypesArray);
+};
+// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+var filterByDay = function(day){
+	return _.filter(foodTrucks, function(foodTruck){
+		if(_.contains(foodTruck.schedule, day)) {
```

Comment:
This check only needs to return true or false. When you return a food truck it interprets it as 'truthy' so it will still work, but it's useful to understand what _.filter is really doing.